### DISCUSSION TITLE
Using 'iptables' in Gradescope for networking assignments

### DISCUSSION THREAD

**Discussion by atul-bansal** [gradescope/autograder_samples#96] [Q&A] [answered]
Hello

I am developing a networking assignment on Gradescope, where I am trying to use 'iptables' tool to govern how incoming and outgoing packets are handled in the docker server. However, after installing iptables and trying to run the command, I get this error - 
`iptables v1.6.0: can't initialize iptables table 'filter': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.`

I looked online to the solution for this issue and they mention using the option `--cap-add=NET_ADMIN` while running the docker image to resolve this. However, I am new to dockers and I am not able to figure out where I can add this option for my autograder.

Till now, I have been using 'zip file upload' option to upload my autograders on Gradescope. Can you help me with this? 

Thank you in advance.

**Comment by ibrahima** [comment, marked as answer]
Hi @atul-bansal, unfortunately that would require elevated privileges on our Docker cluster, so it's not something that you are able to do on our platform at the moment. I think some people have had success with full-system emulation via `qemu` or other emulators, because if you run an emulated system within Docker you can actually have full root access there and do whatever you want with network interfaces and the like. There will be a performance penalty but it may be sufficient for your needs.
